### ISSUE TITLE
bike should better avoid primary roads

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#3008]
E.g. [this route](https://graphhopper.com/maps/?point=52.479741%2C13.346185&point=52.48804%2C13.370555&profile=bike) should not happen but it uses the primary road, even [when forcing](https://graphhopper.com/maps/?point=52.479741%2C13.346185&point=52.481408%2C13.350262&point=52.48804%2C13.370555&profile=bike) with a via point.

The reason here are the tags `maxspeed=30` and `cycleway:right=share_busway` for the primary road which overrule the avoidance.

**Comment by vizsim** [comment]
funfact: In the meantime a real cycleway has been built on that example route:
https://www.mapillary.com/app/?lat=52.4846613&lng=13.353633400000035&z=18.83520815313215&pKey=835943008414001&dateFrom=2024-04-01&x=0.3843274054032111&y=0.5544758255591679&zoom=0&focus=photo

It is now tagged with `cycleway:right = lane`:
https://www.openstreetmap.org/way/508614988#map=18/52.485210/13.354627

It would be really amazing to be able to add those bicycle infrastructure informations (f.e. for `cycleway:right = lane` or `cycleway:right=share_busway` ) as Encoded Values/"Road attribute". 
It would boost the possibilities for custom bicycle routing and would also allow better debugging for unexpected behaviour.
Once it is stored as Road attribute, it would be possible to vizualize the Route segments like this, right?
![grafik](https://github.com/user-attachments/assets/6692f3e9-0502-4750-a6b8-27a44e3c2697)

Or is there an existing workaround for this?

**Comment by ratrun** [comment]
> t would be really amazing to be able to add those bicycle infrastructure informations (f.e. for `cycleway:right = lane` or `cycleway:right=share_busway` ) as Encoded Values/"Road attribute".
> It would boost the possibilities for custom bicycle routing and would also allow better debugging for unexpected behaviour.

I don't understand it. Could you provide a custom rules examples and explain in more detail? 

I believe that merging of #3039 will solve this issue.

**Comment by vizsim** [comment]
I imagine a "Road attribute" like "cycling_infrastucture" that would store values like "cycleway_lane" in it, like in this mock up:
![grafik](https://github.com/user-attachments/assets/e7976b22-b92d-40a2-b9ad-a0123305977b)


I have been trying to explain the "issue" with non-separated cycling infrastucture here:
https://discuss.graphhopper.com/t/take-cycleway-s-along-roads-into-account-for-custom-routing-profiles/9184
and here:
https://discuss.graphhopper.com/t/unexpected-routing-by-bike-ignoring-cycle-paths/9199

I am super happy to elaborate on this, if you need more further details. In here I just realized, that the vizualisation (graph on the bottom right) would help to understand what the routing actually does.

**Comment by ratrun** [comment]
@vizsim Thanks for the visualization. I think now I understood the basic idea, but I'm not sure about the details. If I got it right you want to derive this new encoded value from the [cycleway tag](https://wiki.openstreetmap.org/wiki/Key:cycleway) and the value would be be mainly for shared ways. What value would you assign for separated ways, e.g. a `highway=cycleway`? Would this be "MISSING"?

**Comment by michaz** [comment]
> The reason here are the tags maxspeed=30 and cycleway:right=share_busway for the primary road which overrule the avoidance.

Another reason is that the alternative route you are proposing has segments of cobblestone, which also overrules the avoidance, both in GraphHopper and reality.

Roads like this are not **nice**, but few Berlin commuters take a detour through cobblestones to avoid them.

**Comment by michaz** [comment]
Shall we close here, if only because there is a bike lane by now? :-) 

@vizsim I think you're right. "Does this road have a separate bike lane" is the single most important thing I want to know when I'm customizing my weights, right? First of all to exempt that road from penalties that would come from traffic. 

(That's what's currently happening, too. It's just not available to the `custom_model`.)

One problem (or maybe just "issue"?) is that, like I think @ratrun is saying, a "separate bike lane" can be mapped as two very different things:
- a tag on the road
- an actual separate way next to the road that is *only* the bike lane

In theory, the router should macroscopically always do the same thing in both cases, meaning a road including its bike lane should never get a bonus or penalty just for being mapped this way or the other.

Not sure that's the case.

**Comment by vizsim** [comment]
I think it would be fine to close this issue.

That said, I’d still really like to see a feature that allows me to (de-)prioritize in the `custom_model` not only separate mapped bicycle infrastructure (`road_class == CYCLEWAY`), but also bicycle infrastructure that is mapped directly on the road. The latter applies to all bike lanes and to `cycleway:*=track` (which represents the same kind of separate cycleway but is mapped in a different way).

In the following Overpass query, you can see the “green” cycleway infrastructure that is currently affected. This type of infrastructure can not be (de-)prioritized in the custom_model in GraphHopper at the moment:
https://overpass-turbo.eu/s/2br6 

Would it make sense to open a new issue specifically for this feature request?

**Comment by ratrun** [comment]
> Would it make sense to open a new issue specifically for this feature request?

Yes, with a better title. But giving such new encoded value a proper name is not that easy. I'm not happy with the proposed name `cycling_infrastucture` , because here at first glance bicycle parking places come to my mind, which is not what is meant here.
